### PR TITLE
pre-commit

### PR DESCRIPTION
**Pull request by anisfeld**
The code is taken from [here](https://gist.github.com/mattpolicastro/49104afc931d7c281eae) and does not require review.  I added a wiki entry [here](https://github.com/ganong-noel/lab_code/wiki/Automating-lintr-with-git-hooks).

### CODE REVIEW

**Comment by jingxianng** [comment]
@anisfeld I followed the instructions, but git commit didn't throw an error.

**Comment by anisfeld** [comment]
@pascalnoel I'm not sure what happened with @jingxianng. 
Would you like to try this as well? You could either start from the wiki and tell me what is hard to follow or I could give you more explicit instructions that match your exact case.

**Comment by pascalnoel** [comment]
@g-carranza please try to check with analysis/source/test_lintr_file.R which I just committed [here](https://github.com/ganong-noel/strategic/commit/cfe98e6a8f0b7042b0c1e9c8553c74300f1be0f8)

**Comment by g-carranza** [comment]
@pascalnoel @anisfeld 

Lintr also fails for me. 
Screen Shot 2019-06-11 at 10 46 03 AM

**Comment by anisfeld** [comment]
Great! @g-carranza and @pascalnoel I've slightly changed `pre-commit` in a way the internet says works on Windows. I hope it works on mac too. It works on Linux! 

Can you both pull and try it out?

**Comment by g-carranza** [comment]
@anisfeld 
Lintr works for me.
